game/outfits.rpy

```diff
@@ -1,289 +1,2058 @@
+                outfit_dict["accessory"] = self.accessory.reference_name
+
+            if self.necklace and isinstance(self.necklace, JNNecklace):
+                outfit_dict["necklace"] = self.necklace.reference_name
+
+            return json.dumps(outfit_dict)
+
+        def __load(self):
+            """
+            Loads the persisted data for this outfit from the persistent, if it exists.
+            """
+            if store.persistent.jn_outfit_list[self.reference_name]:
+                self.unlocked = store.persistent.jn_outfit_list[self.reference_name]["unlocked"]
+
+        def __save(self):
+            """
+            Saves the persistable data for this outfit to the persistent.
+            """
+            store.persistent.jn_outfit_list[self.reference_name] = self.as_dict()
+
+        def __delete_save(self):
+            """
+            Deletes the persistable data for this outfit from the persistent, if it exists.
+            """
+            if store.persistent.jn_outfit_list[self.reference_name]:
+                del store.persistent.jn_outfit_list[self.reference_name]
+
+        def __filter_outfit(
+            self,
+            unlocked=None,
+            is_jn_outfit=None,
+            not_reference_name=None,
+            has_accessory=None,
+            has_eyewear=None,
+            has_headgear=None,
+            has_necklace=None
+        ):
+            """
+            Returns True, if the outfit meets the filter criteria. Otherwise False.
+
+            IN:
+                - unlocked - the boolean unlocked state to filter for
+                - is_jn_outfit - the boolean is_jn_outfit state to filter for
+                - not_reference_name - list of reference_names the outfit must not have 
+                - has_accessory - the boolean has_accessory state to filter for
+                - has_eyewear - the boolean has_eyewear state to filter for
+                - has_headgear - the boolean has_headgear state to filter for
+                - has_necklace - the boolean has_necklace state to filter for
+
+            OUT:
+                - True, if the outfit meets the filter criteria. Otherwise False
+            """
+            if unlocked is not None and self.unlocked != unlocked:
+                return False
+
+            elif is_jn_outfit is not None and self.is_jn_outfit != is_jn_outfit:
+                return False
+
+            elif not_reference_name is not None and self.reference_name in not_reference_name:
+                return False
+
+            elif has_accessory is not None and bool(self.has_accessory) != has_accessory:
+                return False
+
+            elif has_eyewear is not None and bool(self.has_eyewear) != has_eyewear:
+                return False
+
+            elif has_headgear is not None and bool(self.has_headgear) != has_headgear:
+                return False
+
+            elif has_necklace is not None and bool(self.has_necklace) != has_necklace:
```

Comment:
It's really unclear what's happening here, but since you do a none check, you can remove `bool` conversion.